File: src/UVa/11498/11498.py

'''
*****************************************************************
*                           PROBLEMA                            *
*****************************************************************
*               UVa 11498 - Division of Nlogonia                *
*****************************************************************
'''

def main():
	
	while True:

		k = int(input())
		if k == 0:
			break

		n, m = map(int, input().split())

		for i in range(k):
			x, y = map(int, input().split())

			if (x == n) or (y == m):
				print("divisa")

			elif x > n:
				if y > m:
					print("NE")
				else:
					print("SE")

			else:
				if y > m:
					print("NO")
				else:
					print("SO")


if __name__ == "__main__":
	main()

File: src/UVa/11498/11498.cpp

/*
*****************************************************************
*                           PROBLEMA                            *
*****************************************************************
*               UVa 11498 - Division of Nlogonia                *
*****************************************************************
*/

#include <cstdio>

int main(){

	int n, x, y, a, b;
	scanf("%d",&n);

	while(n!=0){

		scanf("%d %d",&x,&y);
		for(int i=0; i<n; i++){
			scanf("%d %d",&a,&b);
			if(a==x || b==y){
				printf("divisa\n");
			}else{
				if(a>x){
					if(b>y ? printf("NE\n"):printf("SE\n"));
				}else{
					if(b>y ? printf("NO\n"):printf("SO\n"));
				}
			}
		}
		scanf("%d",&n);

	}
	
	return 0;
}
